src/watchful/enricher.py

```diff
@@ -83,7 +82,13 @@ def __init__(self) -> None:
         class.
         """
 
-        pass
+    @abstractproperty
```

Comment:
https://docs.python.org/3/library/abc.html#abc.abstractproperty:~:text=%40abc.abstractproperty-,Deprecated%20since%20version%203.3,-%3A%20It%20is

Reply:
See the `XXX` comment below it: this code exists _solely_ to make `mypy` happy. Currently, there's a code path that references `enrichment_args` but, prior to this patch, `Enricher` didn't have an `enrichment_args`. It still won't work if executed, but this takes us far enough along to address that in a patch that isn't many mechanical changes.